cmd/prometheus/main.go

```diff
@@ -94,13 +97,8 @@ func main() {
 
 	a.HelpFlag.Short('h')
 
-	a.Flag("log.level",
-		"Only log messages with the given severity or above. One of: [debug, info, warn, error, fatal]").
-		Default("info").StringVar(&cfg.logLevel)
-
-	a.Flag("log.format",
-		`Set the log target and format. Example: "logger:syslog?appname=bob&local=7" or "logger:stdout?json=true"`).
-		Default("logger:stderr").StringVar(&cfg.logFormat)
+	a.Flag(promlog.LevelFlagName, promlog.LevelFlagHelp).
```

Comment:
> That would be the simplest and most obvious way to do it, but it's not the only way.

Yes, the alternative is to define some `RegisterFlags` interface that we can pass in. But from your general position I take that you will also object that every component defines a helper type to implement that interface around their flag library.  
Hence, the only solution is that we also wrap our canonical flag package, given us yet more maintenance burden.

One crucial thing that is not being considered (and which could also bite us one day with extracting a bunch of small config types), is that we are offloading things that are part of various components' stability guarantees into a package that is imported by many.  
This means that components that cannot break anything right now, are stuck in a limbo where they cannot update such a package, even with crucial bug/security fixes, because this update may also pull in a breakage.  

The amount of manual(!) effort this requires to not accidentally do this or actually forking/porting partial important fixes temporarily (=~ up to years) is significantly higher than acknowledging that there's a new logging option and adding a flag for it.
The constants we have ensure the important parts of uniformity.

I heard all your arguments, but I will strongly object what's proposed here overall.

>  instead into a labour intensive change.

It's hardly any more effort than updating the vendoring and per my above reasoning, this even should be a manual and conscious step and not a side effect of vendoring.  
The fact that we are blindly trusting those constants for our stable flags is the maximum I'm willing to do.